AchieveIt.API/Extensions/ExceptionMiddlewareExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Net;
+using AchieveIt.Shared.Exceptions;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+
+namespace AchieveIt.API.Extensions
+{
+    public static class ExceptionMiddlewareExtensions
+    {
+        public static void ConfigureExceptionHandler(this IApplicationBuilder applicationBuilder)
+        {
+            applicationBuilder.UseExceptionHandler(appException =>
+            {
+                appException.Run(async context =>
+                {
+                    context.Response.StatusCode = (int) HttpStatusCode.InternalServerError;
```

Comment:
Looks like there is no sense in this middleware since we still will get just internal server error. The same as without middleware.

Reply:
You can have a look at this exception filter https://gist.github.com/kirpichyov/c2c09ea8871495f28b706999d0bc0d14. It can be registered like this:
![image](https://user-images.githubusercontent.com/45832778/150578945-6806857d-704c-430b-8a68-5155339e09a4.png)

Reply:
I don't get it, after making my ExceptionMiddlewareExtensions looks like your example and registering in Startup, my Exception shows without all that configuration. 

![image](https://user-images.githubusercontent.com/65727446/150590147-b707afb9-c4b4-465d-b287-96fccc834635.png)

![image](https://user-images.githubusercontent.com/65727446/150590173-9009be1a-9f32-4d4f-a4f4-095f5fc3f856.png)

Reply:
Answered in pm

Reply:
Done.